sycl/include/sycl/ext/oneapi/matrix/static-query-use.hpp

```diff
@@ -482,6 +482,379 @@ struct matrix_params<
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, unsigned short> && std::is_same_v<Tc, float> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, double> && std::is_same_v<Tc, double> &&
+       (sM == 16 && sN == 16 && sK == 4)))
+    return true;
+  else
+    return false;
+}
+
+template <typename Ta, typename Tc>
+constexpr bool are_types_valid_amd_gfx90a() {
+  if ((std::is_same_v<Ta, half> && std::is_same_v<Tc, float>) ||
+      (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t>) ||
+      (std::is_same_v<Ta, unsigned short> && std::is_same_v<Tc, float>) ||
+      (std::is_same_v<Ta, double> && std::is_same_v<Tc, double>))
+    return true;
+  else
+    return false;
+}
+
+// Default-values query:
+// Specialization for when only types are given, need to query only sizes
+template <typename Ta, typename Tb, typename Tc, typename Td>
+struct matrix_params<
+    architecture::amd_gpu_gfx90a, Ta, Tb, Tc, Td, 0, 0, 0,
+    typename std::enable_if<(
+        !std::is_same_v<Ta, void> && !std::is_same_v<Tb, void> &&
+        !std::is_same_v<Tc, void> && !std::is_same_v<Td, void> &&
+        std::is_same_v<Ta, Tb> && std::is_same_v<Tc, Td>)>::type> {
+
+  static_assert(
+      (are_types_valid_amd_gfx90a<Ta, Tc>()),
+      "Invalid types for AMD gfx90a, supported types are half, float, "
+      "int8_t, int32_t, double and bf16 (Note that unsigned short"
+      "should be used in the DPC++ code to implement bf16) ");
+
+  // Default sizes for AMD gfx90a were chosen to represent a square matrix
+  static constexpr std::size_t M = 16;
+  static constexpr std::size_t N = 16;
+  static constexpr std::size_t K = ((sizeof(Ta) == 8) ? 16 : 4);
+
+  template <typename Group, layout Layout>
+  using joint_matrix_a = joint_matrix<Group, Ta, use::a, M, K, Layout>;
+  template <typename Group, layout Layout>
+  using joint_matrix_b = joint_matrix<Group, Tb, use::b, K, N, Layout>;
+  template <typename Group>
+  using joint_matrix_c = joint_matrix<Group, Tc, use::accumulator, M, N>;
+  template <typename Group>
+  using joint_matrix_d = joint_matrix<Group, Td, use::accumulator, M, N>;
+};
+
+// Validation query
+// Specialization when both types and sizes are given
+template <typename Ta, typename Tb, typename Tc, typename Td, size_t sM,
+          size_t sN, size_t sK>
+struct matrix_params<
+    architecture::amd_gpu_gfx90a, Ta, Tb, Tc, Td, sM, sN, sK,
+    typename std::enable_if<(
+        !std::is_same_v<Ta, void> && !std::is_same_v<Tb, void> &&
+        !std::is_same_v<Tc, void> && !std::is_same_v<Td, void> &&
+        std::is_same_v<Ta, Tb> && std::is_same_v<Tc, Td> && sM != 0 &&
+        sN != 0 && sK != 0)>::type> {
```

Comment:
another instance to potentially use `std::enable_if_t` and also improve it to have no need for below `static_assert`